api/diagnostics.go

```diff
@@ -517,32 +516,11 @@ func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string,
 
 		j.Status = "GET " + fullURL
 		updateSummaryReport("START "+j.Status, node, "", summaryReport)
-		timeout := time.Minute * time.Duration(j.Cfg.FlagDiagnosticsJobGetSingleURLTimeoutMinutes)
-		request, err := http.NewRequest("GET", fullURL, nil)
-		if err != nil {
-			j.Errors = append(j.Errors, err.Error())
-			logrus.Errorf("Could not create a new HTTP request: %s", err)
-			updateSummaryReport(fmt.Sprintf("could not create request for url: %s", fullURL), node, err.Error(), summaryErrorsReport)
-			j.JobProgressPercentage += percentPerURL
-			continue
-		}
-		request.Header.Add("Accept-Encoding", "gzip")
-
-		client := NewHTTPClient(timeout, j.Transport)
-		resp, err := client.Do(request)
+		resp, err := get(client, fullURL)
 		if err != nil {
 			j.Errors = append(j.Errors, err.Error())
-			logrus.Errorf("Could not fetch url %s: %s", fullURL, err)
-			updateSummaryReport(fmt.Sprintf("could not fetch url: %s", fullURL), node, err.Error(), summaryErrorsReport)
-			j.JobProgressPercentage += percentPerURL
-			continue
-		}
-
-		if resp.StatusCode != http.StatusOK {
-			errMsg := fmt.Sprintf("unable to fetch %s. Return code %d", fullURL, resp.StatusCode)
-			logrus.Error(errMsg)
-			j.Errors = append(j.Errors, errMsg)
-			updateSummaryReport("wrong status code returned", node, errMsg, summaryErrorsReport)
+			logrus.Errorf("Could not read force-tls flag: %s", err)
```

Comment:
this blanket error message doesn't look quite right

Reply:
Good point